### PR TITLE
Added screenshots

### LINKED ISSUE #25: Screencat and screenshots
Follow-up of https://github.com/rugk/unicodify/issues/20

We need:

## One screencast

IMHO, the best to show is how the add-on can transform text.
So just make one with a nice (realistic) example text that could show a good use case for the add-on. (Likely a social media post or so… where I’d prefer Mastodon or so. :sweat_smile: )

Well… thinking about it, we can combine both big features by auto-correcting text first and then transforming some text parts.

## Some screenshots

* some screenshots of auto-corrected text, maybe so it is obvious that the autocorrection happened … or… if really needed… some “before / after” combined in one screenshot (I’d avoid it, however)
* and a screenshot of all text transform options
* showing the options completely

### PR DESCRIPTION
**Pull request by tdulcet**
Fixes #25 (screenshots only).

~Four of the Thunderbird screenshots will need to be added in a follow up PR because of https://github.com/rugk/unicodify/issues/25#issuecomment-843165128.~

The issue in https://github.com/rugk/unicodify/pull/21#pullrequestreview-646710403 is showing in some of the screenshots of the options page. I also had to modify my local copy of the extension to workaround https://github.com/TinyWebEx/AutomaticSettings/issues/11.

I was unsure of the expected format and use of the `amoScreenshots.csv` file, as your example in https://github.com/rugk/unicodify/issues/25#issuecomment-840626954 is an invalid CSV file, as it is using semicolons instead of commas. The name implies that it is only for AMO, although I listed all the screenshots in it.

Please let me know if you would like me to redo any of the screenshots.

### CODE REVIEW

**Comment by rugk** [comment]
> is an invalid CSV file, as it is using semicolons instead of commas. The name implies that it is only for AMO, although I listed all the screenshots in it.

Well… semicolons are also a common separator in CSVs, but yeah, it does not matter since it is never automaticaly parsed. YOu actually need to manually copy it into the AMO description. :upside_down_face: 

Anyway, thanks a lot and I’ll have a look/review that soon (please ping me/request a review if I should forget it).

File: assets/screenshots/amoScreenshots.csv [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+optionsPageFirefox.png, "Options Page, Firefox"
+optionsPageThunderbird.png, "Options Page, Thunderbird"
+textTransformOptionsChrome.png, "Text Transform options, Light theme, Chrome"
+textTransformOptionsDarkChrome.png, "Text Transform options, Dark theme, Chrome"
+textTransformOptionsDarkFirefox.png, "Text Transform options, Dark theme, Firefox"
+textTransformOptionsFirefox.png, "Text Transform options, Light theme, Firefox"
+textTransformOptionsLivePreviewDarkFirefox.png, "Text Transform options with Live preview, Dark theme, Firefox"
+textTransformOptionsLivePreviewFirefox.png, "Text Transform options with Live preview, Light theme, Firefox"
+textTransformOptionsLivePreviewDarkThunderbird.png, "Text Transform options with Live preview, Dark theme, Thunderbird"
+textTransformOptionsLivePreviewThunderbird.png, "Text Transform options with Live preview, Light theme, Thunderbird"
```
Comment (rugk) [inline assets/screenshots/amoScreenshots.csv:14]: As for the text here, don’t be so objective… :stuck_out_tongue_winking_eye: 

This is no image description that just needs to describe what you can already see, but should be some marketing thing, some “fluffy” words. Again [see our Offline QR Code example](https://github.com/rugk/offline-qr-code/blob/master/assets/texts/en/amoScreenshots.csv).
It’s basically about highlighting/presenting nice features and at best also describing boring pages like the options page in a positive way that makes users think “This is cool/special/I want that.”, you see… Try to invoke emotions, (as far as that’s possible) but of course also stay honest and exact. :smile: It should also be quite brief, as it should be easy and fast to read.

E.g. for the settings:
> Disable some features you do not use or just make use of all of them. You can adjust the style and behaviour as you like!

etc. etc.

(If the image only differs in being dark or light of course just use the same description, we’ll likely only use/upload one version as said… it makes no sense to display all of them in all styles.)

If you are out of ideas, I can also look at the other’s and think of some descriptions that would fit, if you want. :wink:

**Review by rugk — CHANGES_REQUESTED** [review]
* as for `optionsPageChrome.png` is there any way to make it show the full page somehow? Using the web dev tools to create a screenshot of all options?
  Or use the dev tools to make your screen 10k pixel height or so, so the 
  Or if nothing else works, maybe manually open the URL of the add-on and `/options/options.html` (though that is maybe bad as you loose the Chrome UI then)
  The use case of that screenshot is after all so that users can see all options in advance (before installing the extension) – it’s not good if they only see the first X ones.
* As for the options, did you really choose the default options? IIRC live preview is not enabled by default, but in your screenshots it is. Or am I wrong?
* Nice, good idea to include the casing options!
* Uhm funny how in Chrome/ium some squared letters are red. Do you think this warrants a new issue here? After all, it's confusing at least… (but maybe also hard to work around…)
* As for the first sentence I would have ended it with “them… how awesome!” or so instead of “them… great” as "the term “great” is already used in the sentence before. However, it’s okay, so you _don’t need to recreate the screenshots just for that_, only if you mind/anyway change them, you could include it! :smiley: 
  
Apart from that looks really cool and thanks again for doing it! :tada:

File: assets/screenshots/amoScreenshots.csv [hunk 2]
```diff
@@ -0,0 +1,16 @@
+autocorrectedTextChrome.png, "Autocorrected Text, Chrome"
```
[the remark this answers is not in the document]
Comment (tdulcet) [reply assets/screenshots/amoScreenshots.csv:1]: OK, will do.

File: assets/screenshots/amoScreenshots.csv (hunk 1 above)
Comment (tdulcet) [reply assets/screenshots/amoScreenshots.csv:14]: > If the image only differs in being dark or light of course just use the same description, we’ll likely only use/upload one version as said… it makes no sense to display all of them in all styles.

OK, I was only following your suggestion in https://github.com/rugk/unicodify/issues/25#issuecomment-841317348.

> If you are out of ideas, I can also look at the other’s and think of some descriptions that would fit, if you want. 😉

Please feel free to suggest other descriptions! There are of course only four unique images (without the browser and theme variations), so what about these descriptions:

1. Autocorrect more than 85 Unicode symbols as you type.
2. Transform selected text to more than 20 Unicode fonts or change the casing.
3. Show a live preview of all the text transform options.
4. Disable any features you do not want or just make use of them all. You can adjust the style and behavior as you like!

You are definitely much better at generating/writing them! 😀

**Comment by tdulcet** [comment]
> * as for `optionsPageChrome.png` is there any way to make it show the full page somehow?

Sure, see https://github.com/TinyWebEx/AutomaticSettings/issues/12#issuecomment-812993836. Your AutomaticSettings library would likely need to add some CSS to specify a width for the options page and possibility a height. This can also be an issue with popups in Chrome. See [here](https://github.com/tdulcet/Colab-Autorun-and-Connect/blob/8c8882b2355f6d659c812a9e5d9cdc21b441d0b9/popup.css#L3-L5) for the CSS I used for my Colab Autorun and Connect add-on. I can of course add some CSS to my development copy of this add-on, although I figured we wanted the screenshots to show what it would actually look like for users.

> * As for the options, did you really choose the default options? IIRC live preview is not enabled by default, but in your screenshots it is. Or am I wrong?

Oops, looks like I forgot to reset the settings. I will retake those two screenshots.

> * Uhm funny how in Chrome/ium some squared letters are red. Do you think this warrants a new issue here? After all, it's confusing at least… (but maybe also hard to work around…)

Yeah, see https://github.com/rugk/awesome-emoji-picker/pull/93#issuecomment-710780555. Several of the letters for a couple of the Unicode fonts (see [here](https://en.wikipedia.org/wiki/Enclosed_Alphanumeric_Supplement#Emoji) and [here](https://en.wikipedia.org/wiki/Enclosed_Alphanumerics#Emoji)) are also emojis and Chrome decided to render them as such. I do not think there is much we can do about it.

> * As for the first sentence I would have ended it with “them… how awesome!” or so instead of “them… great” as "the term “great” is already used in the sentence before. However, it’s okay, so you _don’t need to recreate the screenshots just for that_, only if you mind/anyway change them, you could include it! 😃

OK. I tried to select that existing “great”, but the context menu would cover almost all the text, so I just added the new “great” at the end for consistency, so all the screenshots of the text transformation feature would still use the word “great”. I obviously could not select any of the existing words in the second sentence, as they were all already transformed. Adding “how awesome!” instead is a good idea, so I will do that if I need to redo those screenshots…

File: assets/screenshots/amoScreenshots.csv (hunk 1 above)
Comment (rugk) [reply assets/screenshots/amoScreenshots.csv:14]: Well…ł you are not bad either and I like them quite much.

As for:
```diff
- Show a live preview of all the text transform options.
+ Show a live preview of all the text transform options directly in the context menu.
```

(to highlight where it is shown)

Anything else sounds very good as far as I see.

Ah only this:
```diff
- Transform selected text to more than 20 Unicode fonts or change the casing.
+ Transform the selected text to more than 20 Unicode fonts or change the casing.
```

(I guess that’s grammatically more correct :wink: )

**Comment by tdulcet** [comment]
> Four of the Thunderbird screenshots will need to be added in a follow up PR because of [#25 (comment)](https://github.com/rugk/unicodify/issues/25#issuecomment-843165128).

I found a workaround for this, so I could at least take the screenshots.

**Comment by rugk** [comment]
Sure, do what you think is a good idea.

File: assets/screenshots/amoScreenshots.csv [hunk 3]
```diff
@@ -0,0 +1,16 @@
+autocorrectedTextChrome.png, "Show a live preview of all the text transform options directly in the context menu."
+autocorrectedTextFirefox.png, "Show a live preview of all the text transform options directly in the context menu."
+autocorrectedTextThunderbird.png, "Show a live preview of all the text transform options directly in the context menu."
+optionsPageChrome.png, "Disable any features you do not want or just make use of them all. You can adjust the style and behavior as you like!"
```
Comment (rugk) [inline assets/screenshots/amoScreenshots.csv:4]: when re-reading/proofreading it: `make use of them all` – uhm is this proper English?

should not it say `make use of all of them`?

**Review by rugk — CHANGES_REQUESTED** [review]
Images LGTM. :smiley:

File: assets/screenshots/amoScreenshots.csv (hunk 3 above)
Comment (tdulcet) [reply assets/screenshots/amoScreenshots.csv:4]: Yeah, will do.

**Comment by tdulcet** [comment]
I think this PR is the last piece needed for us to release v0.5, since #4 is obviously not going to get fixed for a while. #36 and #43 included some important bug fixes we should get published as soon as possible.

However, one thing I noticed when taking the screenshots of the options page is that the RandomTips are still not working: 
image
We probably should reopen #23. There are also a couple other issues listed in the third paragraph at the top of the PR that we probably should fix first…

**Comment by rugk** [comment]
> There are also a couple other issues listed in the third paragraph at the top of the PR that we probably should fix first… 

Sure, would you mind to create issues for them? Feel free to assign me if it is related to the RandomTips lib, so I’ll fix it then.

**Pull request merged**